src/aliceVision/calibration/checkerDetector_io.hpp

```diff
@@ -0,0 +1,117 @@
+// This file is part of the AliceVision project.
+// Copyright (c) 2022 AliceVision contributors.
+// This Source Code Form is subject to the terms of the Mozilla Public License,
+// v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <aliceVision/calibration/checkerDetector.hpp>
+
+#include <boost/json.hpp>
+
+namespace aliceVision {
+namespace calibration {
+
+/**
+ * @brief Deserialize JSON object to checkerboard corner.
+ */
+CheckerDetector::CheckerBoardCorner tag_invoke(boost::json::value_to_tag<CheckerDetector::CheckerBoardCorner>,
```

Comment:
maybe split in hpp/cpp, they are not template functions, so the implementation can go in a cpp